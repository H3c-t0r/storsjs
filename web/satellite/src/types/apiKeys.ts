// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

/**
 * Exposes all apiKey-related functionality
 */
export interface ApiKeysApi {
    /**
     * Fetch apiKey
     *
     * @returns ApiKey
     * @throws Error
     */
    get(projectId: string): Promise<ApiKey[]>;
    create(projectId: string, name: string): Promise<ApiKey>;
    delete(ids: string[]): Promise<null>;
}

/**
 * ApiKey class holds info for ApiKey entity.
 */
export class ApiKey {
    public id: string;
    public secret: string;
    public name: string;
    public createdAt: string;
    public isSelected: boolean = false;

    constructor(id: string, name: string, createdAt: string, secret: string) {
        this.id = id || '';
        this.name = name || '';
        this.createdAt = createdAt || '';
        this.secret = secret || '';

        this.isSelected = false;
    }

    public formattedName(): string {
        let name = this.name;

        if (name.length < 12) {
            return name;
        }

        return name.slice(0, 12) + '...';
    }

    public getDate(): string {
        if (!this.createdAt) {
            return '';
        }

        return new Date(this.createdAt).toLocaleDateString();
    }
}
