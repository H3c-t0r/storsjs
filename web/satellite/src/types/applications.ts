// Copyright (C) 2024 Storj Labs, Inc.
// See LICENSE for copying information.

import VeeamIcon from '@/assets/apps/veeam.png';
import TrueNASIcon from '@/assets/apps/truenas.png';
import SplunkIcon from '@/assets/apps/splunk.png';
import ArqIcon from '@/assets/apps/arq.png';
import ZertoIcon from '@/assets/apps/zerto.png';
import UnitrendsIcon from '@/assets/apps/unitrends.png';
import S3FSIcon from '@/assets/apps/s3fs.png';
import RucioIcon from '@/assets/apps/rucio.png';
import PhotosPlusIcon from '@/assets/apps/photos+.png';
import MSP360Icon from '@/assets/apps/msp360.png';
import LucidLinkIcon from '@/assets/apps/lucidlink.png';
import IconikIcon from '@/assets/apps/iconik.png';
import HammerspaceIcon from '@/assets/apps/hammerspace.png';
import ElementsIcon from '@/assets/apps/elements.png';
import DockerIcon from '@/assets/apps/docker.png';
import PixelfedIcon from '@/assets/apps/pixelfed.png';
import MastodonIcon from '@/assets/apps/mastodon.png';
import DataverseIcon from '@/assets/apps/dataverse.png';
import CyberDuckIcon from '@/assets/apps/cyberduck.png';
import AtempoIcon from '@/assets/apps/atempo.png';
import AcronisIcon from '@/assets/apps/acronis.png';
import MountainDuckIcon from '@/assets/apps/mountainduck.png';
import FileZillaIcon from '@/assets/apps/filezilla.svg';
import GBLabsIcon from '@/assets/apps/gblabs.png';
import AdSignalIcon from '@/assets/apps/adsignal.png';
import OwnCloudIcon from '@/assets/apps/owncloud.png';
import LivepeerIcon from '@/assets/apps/livepeer.png';
import BunnyCDNIcon from '@/assets/apps/bunnycdn.png';
import CometIcon from '@/assets/apps/comet.png';
import FastlyIcon from '@/assets/apps/fastly.jpg';
import GlobusIcon from '@/assets/apps/globus.png';
import HuggingFaceIcon from '@/assets/apps/huggingface.svg';
import KerberosIcon from '@/assets/apps/kerberos.png';

export enum AppCategory {
    All = 'All',
    FileManagement = 'File Management',
    BackupRecovery = 'Backup & Recovery',
    ContentDistribution = 'Content Distribution',
    CDN = 'CDN',
    AI = 'AI',
    Logs = 'Logs',
    SocialMedia = 'Social Media',
}

export type Application = {
    title: string
    description: string
    category: AppCategory
    src: string
    docs: string
}

export const applications: Application[] = [
    {
        title: 'TrueNAS - iX Systems',
        description: 'Back up a single instance or keep several TrueNAS deployments synchronized using a single source of truth.',
        category: AppCategory.BackupRecovery,
        src: TrueNASIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/ix-systems-truenas',
    },
    {
        title: 'Nebula by GB Labs',
        description: 'Cloud-based media storage solution powered by Storj. Work directly from the GB Labs Cloud with project files that are always in sync.',
        category: AppCategory.FileManagement,
        src: GBLabsIcon,
        docs: 'https://www.storj.io/partner-solutions/gb-labs',
    },
    {
        title: 'Hammerspace',
        description: 'Hammerspace and Storj bring more flexibility and agility to your global data landscape.',
        category: AppCategory.FileManagement,
        src: HammerspaceIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/hammerspace',
    },
    {
        title: 'Ad Signal Match',
        description: 'Deduplicate assets and save on AI costs with Ad Signal Match. Transfer only your unique content to Storj.',
        category: AppCategory.FileManagement,
        src: AdSignalIcon,
        docs: 'https://www.storj.io/partner-solutions/ad-signal',
    },
    {
        title: 'OwnCloud',
        description: 'oCIS, or ownCloud Infinite Scale, is a cutting-edge technology platform for building cloud-native file sync and share applications.',
        category: AppCategory.FileManagement,
        src: OwnCloudIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/ocis',
    },
    {
        title: 'Livepeer',
        description: 'Add live and on-demand video experiences by transcoding videos with Livepeer and storing your media with Storj.',
        category: AppCategory.BackupRecovery,
        src: LivepeerIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/livepeer',
    },
    {
        title: 'Acronis',
        description: 'Reliable backup and disaster recovery solutions for data archiving and organization, seamlessly integrating with Storj.',
        category: AppCategory.BackupRecovery,
        src: AcronisIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/acronis',
    },
    {
        title: 'Iconik',
        description: 'Iconik is easy to use and intuitive with a clean and intuitive platform that organizes your media and makes it searchable.',
        category: AppCategory.FileManagement,
        src: IconikIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/iconik',
    },
    {
        title: 'Photos+',
        description: 'Store and manage your photos and videos in your own cloud storage account.',
        category: AppCategory.FileManagement,
        src: PhotosPlusIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/photos-plus',
    },
    {
        title: 'Veeam',
        description: 'All-in-one backup, recovery, and data security solution that serves both on-premises and cloud storage.',
        category: AppCategory.BackupRecovery,
        src: VeeamIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/veeam',
    },
    {
        title: 'Splunk',
        description: 'Splunk is a data analytics platform that provides data-driven insights across all aspects of a company.',
        category: AppCategory.Logs,
        src: SplunkIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/splunk',
    },
    {
        title: 'Arq Backup',
        description: 'Arq backs up your filesystem with perfect, point-in-time backups of your files.',
        category: AppCategory.BackupRecovery,
        src: ArqIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/arq',
    },
    {
        title: 'Zerto',
        description: 'Easily incorporate immutability into a complete disaster recovery strategy.',
        category: AppCategory.BackupRecovery,
        src: ZertoIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/zerto',
    },
    {
        title: 'Unitrends',
        description: 'Automate manual tasks, eliminate management complexity, and deliver tested hardware and software resilience.',
        category: AppCategory.BackupRecovery,
        src: UnitrendsIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/unitrends',
    },
    {
        title: 'S3FS',
        description: 's3fs allows Linux, macOS, and FreeBSD to mount an S3 bucket via FUSE.',
        category: AppCategory.FileManagement,
        src: S3FSIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/s3fs',
    },
    {
        title: 'Rucio',
        description: 'Organize, manage, and access your data at scale with Rucio, an open-source framework for scientific collaboration.',
        category: AppCategory.FileManagement,
        src: RucioIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/rucio',
    },
    {
        title: 'MSP360',
        description: 'MSP360 is a best-in-class IT management platform for MSPs and internal IT teams.',
        category: AppCategory.BackupRecovery,
        src: MSP360Icon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/msp360',
    },
    {
        title: 'LucidLink',
        description: 'Using Storj with LucidLink provides resilient cloud object storage with blazing performance and zero-trust security.',
        category: AppCategory.FileManagement,
        src: LucidLinkIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/lucidlink',
    },
    {
        title: 'Elements',
        description: 'Store, manage and collaborate on media content using a secure and scalable asset management platform.',
        category: AppCategory.FileManagement,
        src: ElementsIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/elements',
    },
    {
        title: 'Docker',
        description: 'Storj supports custom Content-Type for any key, so it can be used as a container registry to distribute container images.',
        category: AppCategory.ContentDistribution,
        src: DockerIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/docker',
    },
    {
        title: 'Pixelfed',
        description: 'Learn how to set up Pixelfed decentralized social media platform to Storj.',
        category: AppCategory.SocialMedia,
        src: PixelfedIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/pixelfed',
    },
    {
        title: 'Mastodon',
        description: 'Here is how you can set up Mastodon decentralized social media platform to Storj.',
        category: AppCategory.SocialMedia,
        src: MastodonIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/mastodon',
    },
    {
        title: 'Dataverse',
        description: 'S3 credentials allow Dataverse to upload and download files from Storj as if it was using the S3 API.',
        category: AppCategory.FileManagement,
        src: DataverseIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/dataverse',
    },
    {
        title: 'CyberDuck',
        description: 'Open-source cloud storage browser app for macOS, Windows, and Linux that supports Storj.',
        category: AppCategory.FileManagement,
        src: CyberDuckIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/cyberduck',
    },
    {
        title: 'Atempo',
        description: 'Integrate with Storj to reduce storage costs without sacrificing security or performance for media archival with global access.',
        category: AppCategory.BackupRecovery,
        src: AtempoIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/atempo-miria',
    },
    {
        title: 'Mountain Duck',
        description: 'Add Storj as a disk on your computer. Open your Storj files with any application and work like on a local disk.',
        category: AppCategory.FileManagement,
        src: MountainDuckIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/cyberduck',
    },
    {
        title: 'Bunny CDN',
        description: 'Set up a static website hosting with Storj using Bunny CDN as a content delivery network providing a caching layer.',
        category: AppCategory.CDN,
        src: BunnyCDNIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/bunny',
    },
    {
        title: 'Fastly',
        description: 'Distribute your content among the Fastly edge cloud service using your Storj buckets as a source of content.',
        category: AppCategory.CDN,
        src: FastlyIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/fastly',
    },
    {
        title: 'Comet',
        description: 'Flexible backup platform integrates with Storj as a backup and storage destination to protect and restore data.',
        category: AppCategory.BackupRecovery,
        src: CometIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/comet',
    },
    {
        title: 'Globus',
        description: 'Seamless collaboration and management for data transfer, sharing, and discovery, all into one unified open-source platform.',
        category: AppCategory.FileManagement,
        src: GlobusIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/globus',
    },
    {
        title: 'HuggingFace',
        description: 'Train and deploy open-source AI models, while saving and loading datasets to and from Storj.',
        category: AppCategory.AI,
        src: HuggingFaceIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/hugging-face',
    },
    {
        title: 'FileZilla',
        description: 'Learn how to set up FileZilla to transfer files over Storj DCS or integrate FileZilla Pro.',
        category: AppCategory.FileManagement,
        src: FileZillaIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/filezilla',
    },
    {
        title: 'Kerberos.io',
        description: 'Open-source platform for video analytics and management. Integrate with Storj to store your Kerberos Vault video files.',
        category: AppCategory.FileManagement,
        src: KerberosIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/kerberos-vault',
    },
];
