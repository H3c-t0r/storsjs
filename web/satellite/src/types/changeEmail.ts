// Copyright (C) 2024 Storj Labs, Inc.
// See LICENSE for copying information.

export enum ChangeEmailStep {
    InitStep = 0,
    VerifyPasswordStep,
    Verify2faStep,
    VerifyOldEmailStep,
    SetNewEmailStep,
    VerifyNewEmailStep,
    SuccessStep,
}
