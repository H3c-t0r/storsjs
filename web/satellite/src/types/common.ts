// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { Validator } from '@/utils/validation';
import { useConfigStore } from '@/store/modules/configStore';

export enum SortDirection {
    asc = 1,
    desc = 2,
}

export class PricingPlanInfo {
    constructor(
        public type: PricingPlanType = PricingPlanType.FREE,
        // Info for the pricing plan container
        public title: string = '',
        public containerSubtitle: string = '',
        public containerDescription: string = '',
        public containerFooterHTML: string | null = null,
        public activationButtonText: string | null = null,
        // Info for the pricing plan modal (pre-activation)
        public activationSubtitle: string | null = null,
        public activationDescriptionHTML: string = '',
        public activationPriceHTML: string | null = null,
        // Info for the pricing plan modal (post-activation)
        public successSubtitle: string = '',
        public bannerTitle: string = '',
        public bannerText: string = '',
    ) {}
}

export interface OnboardingInfo {
    accessText?: string;
    accessBtnText?: string;
    accessTitle?: string;
}

export enum PricingPlanType {
    FREE = 'free',
    PARTNER = 'partner',
    PRO = 'pro',
}

// TODO: fully implement these types and their methods according to their Go counterparts
export type UUID = string
export type MemorySize = string
export type Time = string

export function tableSizeOptions(itemCount: number, isObjectBrowser = false): {title: string, value: number}[] {
    const opts = [
        { title: '10', value: 10 },
        { title: '25', value: 25 },
        { title: '50', value: 50 },
        { title: '100', value: 100 },
    ];
    if (itemCount <= 300 && !isObjectBrowser) {
        return [{ title: 'All', value: itemCount }, ...opts];
    }
    return opts;
}

export type ValidationRule<T> = string | boolean | ((value: T) => string | boolean);

export function RequiredRule(value: unknown): string | boolean {
    return (Array.isArray(value) ? !!value.length : !!value) || 'Required';
}

export function EmailRule(value: string, strict = false): string | boolean {
    return Validator.email(value, strict) || 'E-mail must be valid.';
}

export function MaxNameLengthRule(value: string): string | boolean {
    const { maxNameCharacters } = useConfigStore().state.config;

    return Validator.nameLength(value, maxNameCharacters) || `The value must be less than or equal to ${maxNameCharacters}.`;
}

export interface IDialogFlowStep {
    onEnter?: () => void;
    onExit?: (to: 'next' | 'prev') => void;
    validate?: () => boolean;
}

export interface DialogStepComponent extends IDialogFlowStep {
    title: string;
    iconSrc?: string;
}

export type SaveButtonsItem = string | {
    name: string;
    value: string;
};

export const MAX_SEARCH_VALUE_LENGTH = 200;

export function getUniqueName(name: string, allNames: string[]): string {
    // Regular expression to match a name with an optional numeric suffix.
    const namePattern = /^(.*?)(?: \((\d+)\))?$/;

    let currName = name;
    let count = 0;
    while (allNames.includes(currName)) {
        count++;
        currName = currName.replace(namePattern, (_, baseName, index) => {
            // Increment the suffix if it exists, or add a new one starting with 1.
            const newIndex = index ? parseInt(index) + 1 : count;
            return `${baseName} (${newIndex})`;
        });
    }

    return currName;
}
