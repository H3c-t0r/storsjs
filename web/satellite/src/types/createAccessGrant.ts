// Copyright (C) 2023 Storj Labs, Inc.
// See LICENSE for copying information.

export enum AccessType {
  APIKey = 'apikey',
  S3 = 's3',
  AccessGrant = 'accessGrant',
}
