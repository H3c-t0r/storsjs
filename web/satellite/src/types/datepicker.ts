// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

export class DateObj {
    constructor(public time: any = null) {}
}

export class Options {
    constructor(
        public type: string = 'multi-day',
        public isSundayFirst: boolean = false,
        public week: string[] = ['Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa', 'Su'],
        public month: string[] = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'],
        public color = {
            checked: '#2683FF',
            header: '#2683FF',
            headerText: '#444C63',
        },
        public inputStyle = {
            'visibility': 'hidden',
            'width': '0',
        },
        public placeholder: string = '',
        public buttons = {
            ok: 'OK',
            cancel: 'Cancel',
        },
        public overlayOpacity: number = 0.5,
        public dismissible: boolean = true,
    ) {}
}
