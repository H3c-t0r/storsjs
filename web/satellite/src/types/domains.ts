// Copyright (C) 2024 Storj Labs, Inc.
// See LICENSE for copying information.

export class Domain {
    constructor(
        public id: string = '',
        public name: string = '',
        public createdAt: Date = new Date(),
    ) { }
}

export enum NewDomainFlowStep {
    CustomDomain = 1,
    SetupDomainAccess,
    EnterNewPassphrase,
    PassphraseGenerated,
    SetupCNAME,
    SetupTXT,
    VerifyDomain,
    DomainConnected,
}
