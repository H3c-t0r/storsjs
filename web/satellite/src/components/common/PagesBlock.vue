// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="pages-container">
        <span
            class="pages-container__pages"
            v-for="page in pages"
            :class="{'selected': isSelected(page.index)}"
            @click="page.select()"
            :key="page.index">{{page.index}}</span>
    </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

import { Page } from '@/types/pagination';

@Component
export default class PagesBlock extends Vue {
    @Prop({default: () => []})
    public readonly pages: Page[];
    @Prop({default: () => false})
    public readonly isSelected: CheckSelected;
}
</script>

<style scoped lang="scss">
    .pages-container {
        display: flex;

        &__pages {
            font-family: 'font_medium';
            font-size: 16px;
            margin-right: 15px;
            width: 10px;
            text-align: center;
            cursor: pointer;
            display: block;
            position: relative;
            transition: all .2s ease;

            &:hover {
                color: #2379EC;

                &:after {
                    content: '';
                    display: block;
                    position: absolute;
                    bottom: -4px;
                    left: 0;
                    width: 100%;
                    height: 2px;
                    background-color: #2379EC;
                }
            }

            &:last-child {
                margin-right: 0;
            }
        }
    }

    .selected {
        color: #2379EC;
        font-family: 'font_bold';

        &:after {
            content: '';
            display: block;
            position: absolute;
            bottom: -4px;
            left: 0;
            width: 10px;
            height: 2px;
            background-color: #2379EC;
        }
    }
</style>
