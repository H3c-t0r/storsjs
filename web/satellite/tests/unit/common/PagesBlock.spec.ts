// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { mount, shallowMount } from '@vue/test-utils';
import * as sinon from 'sinon';
import PagesBlock from '@/components/common/PagesBlock.vue';
import { Page } from '@/types/pagination';

describe('Pagination.vue', () => {
    it('renders correctly without props', () => {
        const wrapper = shallowMount(PagesBlock);

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with props', () => {
        const callbackSpy = sinon.spy();
        let pagesArray: Page[] = [];
        const SELECTED_PAGE_INDEX: number = 3;

        for (let i = 1; i <= 4; i++) {
            pagesArray.push(new Page(i, callbackSpy));
        }

        const wrapper = shallowMount(PagesBlock, {
            propsData: {
                pages: pagesArray,
                checkSelected: (i: number) => i === SELECTED_PAGE_INDEX
            }
        });

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.findAll('span').length).toBe(4);
        expect(wrapper.findAll('span').at(2).classes().includes('selected')).toBe(true);
    });

    it('behaves correctly on page click', async () => {
        const callbackSpy = sinon.spy();
        let pagesArray: Page[] = [];

        for (let i = 1; i <= 3; i++) {
            pagesArray.push(new Page(i, callbackSpy));
        }

        const wrapper = shallowMount(PagesBlock, {
            propsData: {
                pages: pagesArray,
                checkSelected: () => false
            }
        });

        wrapper.findAll('span').at(1).trigger('click');

        expect(callbackSpy.callCount).toBe(1);
    });
});
