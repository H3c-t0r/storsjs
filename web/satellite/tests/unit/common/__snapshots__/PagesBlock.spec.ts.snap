// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination.vue renders correctly with props 1`] = `<div class="pages-container"><span class="">1</span><span class="">2</span><span class="selected">3</span><span class="">4</span></div>`;

exports[`Pagination.vue renders correctly without props 1`] = `<div class="pages-container"></div>`;
